gender_analysis/analysis/corpus_metadata_visualizations.py

```diff
@@ -0,0 +1,209 @@
+import matplotlib.pyplot as plt
+import seaborn as sns
+from collections import Counter
+
+from gender_analysis.common import MissingMetadataError
+
+
+def plt_pubyears(corpus):
+    """
+    Creates a histogram displaying the frequency of books that were published within a 20 year 
+    period
+    Requires that corpus contains a 'date' metadata field
+    :param corpus: Corpus
+    RETURNS a pyplot histogram
+    """
+    if 'date' not in corpus.get_corpus_metadata():
+        raise MissingMetadataError('This corpus does not contain metadata on publication date.')
```

Comment:
Let's make all of these exceptions more explicit: i.e., as is, this exception doesn't explicitly inform the caller that what's missing is a field called `date` in the metadata.

You can actually subclass Exception to take an argument other than just the message if you like -- e.g., you could make an Exception subclass that you can call in this manner:
```
raise MissingMetadataException('Here's a verbose message', 'date')
```
and then the exception would appear something like this:
```
MissingMetadataException - missing metadata field 'date' - 'Here's a verbose message'
```